frame/contracts/src/schedule.rs

```diff
@@ -34,178 +34,194 @@ pub const API_BENCHMARK_BATCH_SIZE: u32 = 100;
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 #[derive(Clone, Encode, Decode, PartialEq, Eq)]
 pub struct Schedule<T: Trait> {
-	/// The type parameter is used in the default implementation.
-	pub phantom: PhantomData<T>,
-
 	/// Version of the schedule.
 	pub version: u32,
 
+	/// The weights for individual wasm instructions.
+	pub instruction_weights: InstructionWeights,
+
+	/// The weights for each imported function a contract is allowed to call.
+	pub api_weights: ApiWeights,
+
+	/// Whether the `seal_println` function is allowed to be used contracts.
+	/// MUST only be enabled for `dev` chains, NOT for production chains
+	pub enable_println: bool,
+
+	/// The maximum number of topics supported by an event.
+	pub max_event_topics: u32,
+
+	/// Maximum allowed stack height.
+	///
+	/// See https://wiki.parity.io/WebAssembly-StackHeight to find out
+	/// how the stack frame cost is calculated.
+	pub max_stack_height: u32,
+
+	/// Maximum number of memory pages allowed for a contract.
+	pub max_memory_pages: u32,
+
+	/// Maximum allowed size of a declared table.
+	pub max_table_size: u32,
+
+	/// The maximum length of a subject used for PRNG generation.
+	pub max_subject_len: u32,
+
+	/// The maximum length of a contract code in bytes. This limit applies to the uninstrumented
+	/// and pristine form of the code as supplied to `put_code`.
+	pub max_code_size: u32,
+
+	/// The type parameter is used in the default implementation.
+	pub _phantom: PhantomData<T>,
+}
+
+/// Describes the weight for all categories of supported wasm instructions.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Clone, Encode, Decode, PartialEq, Eq)]
+pub struct InstructionWeights {
 	/// Weight of a growing memory by single page.
-	pub op_cost_grow_mem: Weight,
+	pub grow_mem: Weight,
 
 	/// Weight of a regular operation.
-	pub op_cost_regular: Weight,
+	pub regular: Weight,
+}
 
+/// Describes the weight for each imported function that a contract is allowed to call.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Clone, Encode, Decode, PartialEq, Eq)]
+pub struct ApiWeights {
```

Comment:
Nit: Not a fan of this name: API is a bit overloaded term. (fwiw, "runtime" is not better xD). Might HostFunctionWeights be better?

Reply:
Can we maybe come up with a custom name for these functions? We have no distinct name to refer to these and it's becoming a problem. Using wasm lingo (host function, imported function) is not good enough because the runtime itself is also using wasm.

So basically a catchy name for these "contract imported functions".

Reply:
I just renamed it to `HostFnWeights`. Should be at least a little better than the current name.